File: Assets/ProjectGame/Lib/data/class_data_id/PlaceMap/PlaceMapTable.py
from ..BaseClassDataID import BaseClassDataID
from .PlaceMapRow import PlaceMapRow
from .PlaceMapTableID import PlaceMapTableID
from typing import Dict

class PlaceMapTable(BaseClassDataID):
    Table: Dict[PlaceMapTableID, PlaceMapRow] = {}

    @classmethod
    def _get_enum(cls, name: str):
        return PlaceMapTableID[name]
    @classmethod
    def _get_row_class(cls):
        return PlaceMapRow

    def read(self, reader):
        self.Table.clear()
        row_count = reader.read_int32()
        col_count = reader.read_int32()
        for _ in range(col_count):
            len_name = reader.read_int32()
            _ = reader.read_string()  # col name（ヘッダー読み飛ばし）
            len_type = reader.read_int32()
            _ = reader.read_string()  # col type
        for _ in range(row_count):
            enum_int = reader.read_int32()
            enum_val = PlaceMapTableID(enum_int)
            row = PlaceMapRow()
            row.read(reader)
            self.Table[enum_val] = row

File: Assets/ProjectGame/Lib/data/class_data_id/PlaceMap/PlaceMapTable.js
import { BaseClassDataID } from '../BaseClassDataID.js';
import { PlaceMapRow } from './PlaceMapRow.js';
import { PlaceMapTableID } from './PlaceMapTableID.js';

export class PlaceMapTable extends BaseClassDataID {
    static Table = new Map();

    static _getEnum(name) {
        return PlaceMapTableID[name];
    }

    static _getRowClass() {
        return PlaceMapRow;
    }

    read(reader) {
        this.constructor.Table.clear();
        const rowCount = reader.readInt32();
        const colCount = reader.readInt32();
        for (let i = 0; i < colCount; i++) {
            const lenName = reader.readInt32();
            reader.readString(); // col name（ヘッダー読み飛ばし）
            const lenType = reader.readInt32();
            reader.readString(); // col type
        }
        for (let r = 0; r < rowCount; r++) {
            const enumInt = reader.readInt32();
            const enumVal = PlaceMapTableID[Object.keys(PlaceMapTableID).find(key => PlaceMapTableID[key] === enumInt)] || enumInt;
            const row = new PlaceMapRow();
            row.read(reader);
            this.constructor.Table.set(enumVal, row);
        }
    }

    static getRow(id) {
        return this.Table.get(id) ?? null;
    }
}
